ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/checkboxgroup/v1/checkboxgroup/clientlibs/site/js/checkboxgroupview.js

```diff
@@ -115,6 +121,81 @@
             }, this)
             super.updateEmptyStatus();
         }
+        #createCheckBoxItem(value, itemLabel) {
+            let option = document.createElement('div');
+            option.classList.add(CheckBoxGroup.selectors.item.slice(1));
+            let label = document.createElement('label');
+            label.classList.add(CheckBoxGroup.selectors.optionLabel.slice(1));
+
+            let input = document.createElement('input');
+            input.type = 'checkbox';
+            input.classList.add(CheckBoxGroup.selectors.widget.slice(1));
+            input.value = value;
+
+            let span = document.createElement('span');
+            span.textContent = itemLabel;
+
```

Comment:
Line 125-136 need to be templatized, rather than creating this via code. This will enable better experience for customization

Reply:
@sakshi-arora1 
Can you point to some example of where we have done this kinda templating before?

Reply:
done!